### PR TITLE
fix(control_validator): resolve the bug causing inappropriate diagnostic messages.

### PR DESCRIPTION
**Pull request by keiota**
## Description

In this PR, I resolved a bug showing inappropriate messages "planning trajectory is too far from ego!!", that occur during deviation check between reference trajectory and predicted trajectory.

The cause of this bug was that empty coordinates were added to the trajectory and the wrong length array was provided.

### Before this PR
During Psim, after decided a rooting we can constantly see a warning message "planning trajectory is too far from ego!!" even though the trajectory looks no problem.
```[component_container-34] [WARN] [1693455073.217942402] [control.control_validator]: planning trajectory is too far from ego!!```

### After this PR
I have not tested it yet, but this message no longer appears incorrectly.

## Tests performed

I haven't test with unit test but checked on my computer that message is not shown.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/control_validator/src/utils.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ void insertPointInPredictedTrajectory(
 
 TrajectoryPoints reverseTrajectoryPoints(const TrajectoryPoints & trajectory_points)
 {
-  TrajectoryPoints reversed_trajectory_points(trajectory_points.size());
+  TrajectoryPoints reversed_trajectory_points(0);
```
Comment (kyoichi-sugahara) [inline control/control_validator/src/utils.cpp:47]: @kei-t4 
thank you for detecting my stupid bug.
I think what I wanted to do is here.
```suggestion
  TrajectoryPoints reversed_trajectory_points;
  reversed_trajectory_points.reserve(trajectory_points.size());
```

And setting the vector size to 0 in the constructor is unnecessary, as the default constructor already creates an empty vector.

Comment (kei-t4) [reply control/control_validator/src/utils.cpp]: @kyoichi-sugahara 

You are mentioning to me in this message, but I think it should be not me, but should someone you work with.
Please check the message recipient. 
Thanks.

**Comment by kyoichi-sugahara** [comment]
@keiota 
Thanks to this change I confirmed error is fixed and control output is evaludated propely thanks!
Screenshot from 2023-09-03 23-15-43
Screenshot from 2023-09-03 23-14-09

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!
Thank you so much for your contribution!

**Pull request merged**